src/pybind/mgr/dashboard/frontend/src/app/ceph/new-dashboard/dashboard/dashboard.component.ts

```diff
@@ -159,56 +161,48 @@ export class DashboardComponent implements OnInit, OnDestroy {
 
   triggerPrometheusAlerts() {
     this.prometheusService.ifAlertmanagerConfigured(() => {
-      this.isAlertmanagerConfigured = true;
-
       this.prometheusService.getAlerts().subscribe((alerts) => {
         this.alerts = alerts;
-        this.crticialActiveAlerts = alerts.filter(
-          (alert: AlertmanagerAlert) =>
-            alert.status.state === 'active' && alert.labels.severity === 'critical'
-        ).length;
-        this.warningActiveAlerts = alerts.filter(
-          (alert: AlertmanagerAlert) =>
-            alert.status.state === 'active' && alert.labels.severity === 'warning'
-        ).length;
       });
     });
   }
 
   getPrometheusData(selectedTime: any) {
-    if (this.timerGetPrometheusDataSub) {
-      this.timerGetPrometheusDataSub.unsubscribe();
-    }
-    this.timerGetPrometheusDataSub = timer(0, this.timerTime).subscribe(() => {
-      selectedTime = this.updateTimeStamp(selectedTime);
-
-      for (const queryName in queries) {
-        if (queries.hasOwnProperty(queryName)) {
-          const query = queries[queryName];
-          let interval = selectedTime.step;
-
-          if (query.includes('rate') && selectedTime.step < 20) {
-            interval = 20;
-          } else if (query.includes('rate')) {
-            interval = selectedTime.step * 2;
+    this.prometheusService.ifPrometheusConfigured(() => {
+      if (this.timerGetPrometheusDataSub) {
+        this.timerGetPrometheusDataSub.unsubscribe();
+      }
+      this.timerGetPrometheusDataSub = timer(0, this.timerTime).subscribe(() => {
+        selectedTime = this.updateTimeStamp(selectedTime);
+
+        for (const queryName in queries) {
+          if (queries.hasOwnProperty(queryName)) {
+            const query = queries[queryName];
+            let interval = selectedTime.step;
+
+            if (query.includes('rate') && selectedTime.step < 20) {
+              interval = 20;
+            } else if (query.includes('rate')) {
+              interval = selectedTime.step * 2;
+            }
+
+            const intervalAdjustedQuery = query.replace(/\[(.*?)\]/g, `[${interval}s]`);
+
+            this.prometheusService
+              .getPrometheusData({
+                params: intervalAdjustedQuery,
+                start: selectedTime['start'],
+                end: selectedTime['end'],
+                step: selectedTime['step']
+              })
+              .subscribe((data: any) => {
+                if (data.result.length) {
+                  this.queriesResults[queryName] = data.result[0].values;
+                }
+              });
```

Comment:
Most of this code is unrelated to the Dashboard presentation and rather about processing Prometheus data, so why not having this encapsulated in the proper Prometheus service?

Not sure how 'customized' the queries are, but if they are simple queries, we could [try and perhaps fetch multiple timeseries in a single query](https://stackoverflow.com/questions/57627673/how-to-send-multiple-queries-to-prometheus-http-api).

Reply:
@Pegonzal since you worked the most here I will need your help with this. What do you think about this?

Reply:
> Most of this code is unrelated to the Dashboard presentation and rather about processing Prometheus data, so why not having this encapsulated in the proper Prometheus service?

I could look into cleaning it up a bit, and refactor it into the prometheus service.

> Not sure how 'customized' the queries are, but if they are simple queries, we could [try and perhaps fetch multiple timeseries in a single query](https://stackoverflow.com/questions/57627673/how-to-send-multiple-queries-to-prometheus-http-api).

I was looking into it and it might work, thanks Ernesto. I'll try to have them all together into one query to optimize it.

If you want @nizamial09 we can create a tracker and I'll take care of it, on a separate PR.

Reply:
yup, i'll open a separate tracker and will assign it to you,

Reply:
https://tracker.ceph.com/issues/58936